### PR TITLE
129-ui-manage-permissions

### PR DESCRIPTION
**Pull request by Tguntenaar**
_Resolves #129_

Still a WIP
[Testnet preview](https://test.near.social/geforcy.testnet/widget/gigs-board.pages.Teams)
[Mainnet preview](https://near.social/geforcy.near/widget/gigs-board.pages.Teams)

Acceptance criteria:
- [x] When non moderator user logs in, they cannot see or access the edit permissions page
- [x] When moderator user logs in, there is a button in the Teams page, to enter edit mode
    - [x] Moderators can add and remove restricted labels 
    - [x] Moderators can create and delete teams
    - [x] Moderators can add member and remove member from a team
    - [x] Moderators can specify which team can do with which restricted label
    - [x] Moderators can specify which team has which restricted label

### CODE REVIEW

**Comment by ori-near** [comment]
Hi @Tguntenaar – love the way you're tracking the AC in the PR. Cool idea. Maybe we should add this to our process ...

**Comment by Tguntenaar** [comment]
>  Moderators can specify which team can do what with which restricted label

@ori-near Are you asking whether moderators have the ability to restrict their actions to editing posts with specific labels, as well as restrict assigning and unassigning ONLY for specific labels.

Hopefully my question is clear. As of right now a moderator can specify one or multiple rules/labels for teams however these labels always give _and_ editing rights _and_ assigning rights. So my question is: should this be: **AND/OR** instead of **AND**.

**Comment by Tguntenaar** [comment]
Hey guys!

I added you guys as moderator in the mainnet preview. Please let me know your testnet handles if you wish to test it on the testnet preview.

Steps I followed to test the functionality added:
- [ ] Add a restricted label
- [ ] Add a restricted label that starts with "start-with:"
- [ ] Add a team
- [ ] Add a member to that team
- [ ] Edit the permission of the label to be either 'edit-post' or 'use-labels'
- [ ] Remove a label from the team
- [ ] Add a label to the team
- [ ] Remove a member from the team
- [ ] Remove a team
- [ ] Remove a restricted label

Finally I have repeated these steps as moderator

File: src/gigs-board/entity/team/LabelsPermissions.jsx [hunk 1]
```diff
@@ -155,19 +157,45 @@ const permissionExplainer = (permission) => {
...
+  txn.push({
+    contractName: nearDevGovGigsContractAccountId,
+    methodName: "unset_restricted_rules",
+    args: {
+      rules: [name],
+    },
+    deposit: Big(0).pow(21),
+    gas: Big(10).pow(12).mul(100),
+  });
+  Near.call(txn);
```
Comment (ailisp) [inline src/gigs-board/entity/team/LabelsPermissions.jsx:171]: Can be simplified to `Near.call({contractName: ...)`

**Review by ailisp — APPROVED** [review]
Looks great on my side! It's a lot of functionality impleneted and testing, @Tguntenaar great job! @ori-near Is there any additional concerns?

**Review by ailisp — APPROVED** [review]
The functionality looks good. As we discussed today, we'll merge this and improve the UX in a future ticket.

**Pull request merged**